app/controllers/key_pairs_controller.rb

```diff
@@ -114,47 +115,53 @@ def create
     time_stamp = Time.now.to_i.to_s
 
     public_key_content = ''
-
+    # If user uploads the public key as a file then that takes precedence over
+    # the key_string
     if !key_pair_params[:file]
-      # Dump the string into the file
+      # Get key from key_string param
       public_key_content = key_pair_params[:key_string]
     else
+      # Get key from file contents
       public_key_content = key_pair_params[:file].read
     end
 
-    # If user uploads the public key as a file then that takes precedence over
-    # the key string
-    if !key_pair_params[:file]
-      # Create a .pub file on the file system
-      upload_key_file(public_key_content, time_stamp)
-    else
+    # Check to see if the public_key_content is a valid ssh key: an ssh
+    # key has the format "type blob email" and cannot have a nil type or blob.
+    type, blob, _email = public_key_content.split
+    if !type.nil? && !blob.nil?
       # Upload the file
       upload_key_file(public_key_content, time_stamp)
-    end
 
-    # Save the record
-    @key_pair = KeyPair.new(
-      key_pair_params.merge(user_name: @current_user.user_name,
-                            user_id: @current_user.id,
-                            file_name: @current_user.user_name +
-                                '@' + time_stamp + '.pub'))
-
-    respond_to do |format|
-      if @key_pair.save
-        format.html do
-          redirect_to key_pairs_path,
-                      notice: 'Key pair was successfully created.'
+      # Save the record
+      @key_pair = KeyPair.new(user_name: @current_user.user_name,
+                              user_id:   @current_user.id,
+                              file_name: @current_user.user_name +
+                                           "@#{time_stamp}.pub")
+
+      respond_to do |format|
+        if @key_pair.save
+          format.html do
+            redirect_to key_pairs_path,
+                        notice: 'Key pair was successfully created.'
+          end
+          format.json do
+            render json: @key_pair,
+                   status: :created,
+                   location: @key_pair
+          end
+        else
+          format.html { render action: 'new' }
+          format.json do
+            render json: @key_pair.errors,
+                   status: :unprocessable_entity
+          end
         end
-        format.json do
-          render json: @key_pair,
-                 status: :created,
-                 location: @key_pair
-        end
-      else
-        format.html { render action: 'new' }
-        format.json do
-          render json: @key_pair.errors,
-                 status: :unprocessable_entity
+      end
+    else # if type and/or blob are nil
+      respond_to do |format|
+        format.html do
+          redirect_to :back,
+                      flash: { error: 'Key pair was not created: Invalid key.' }
```

Comment:
Two comments:  First, please set up the message using flash_message (from helpers/application_helper.rb).  We are trying to move towards using this helper function more.  Second, the text of the message itself should go in the language files.  You'll need to add a key for it.